### DISCUSSION TITLE
spacy visualizer for the dependency parser

### DISCUSSION THREAD

**Discussion by zarakolagar120** [explosion/spaCy#8642] [Help: Other Questions] [answered]
Hi,
I have a question regarding the spacy visualizer for the dependency parser.
I was wondering how do you guys create the arcs information for the visualizer, is that information coming from the dependency model along with the dependency tags, or have you trained another model to output arcs based on the dependency and/ or POS models' output? 
and if this info is coming from such a model, is this model finetuneable? 
thank you very much in advance.

**Comment by polm** [comment, marked as answer]
A dependency parser labels tokens with their parent (another token, or ROOT) and label (from a fixed list). The arcs are just drawn to connect each token to its parent. So the information is just coming from the dependency parser.

You can train your own dependency parser the same as any other spaCy model.
